onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -125,7 +129,10 @@ def optimize_by_onnxruntime(
         sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_ALL
 
     if optimized_model_path is None:
-        path_prefix = onnx_model_path[:-5]  # remove .onnx suffix
+        if isinstance(onnx_model, str):
+            path_prefix = onnx_model[:-5]  # remove .onnx suffix
```

Comment:
Is the model path always going to end in `.onnx`? Will this fail when the model file does not have `.onnx` extension? I suggest using `pathlib` for file path handling.

Reply:
I think so, otherwise this method will even fail to load the model?

Reply:
Model loading does not assume file extensions afak. It will treat the file as a protobuf message in any case.